pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -66,6 +70,21 @@ type topologyPairsMaps struct {
 	podToTopologyPairs map[string]topologyPairSet
 }
 
+// topologyPairsPodSpreadMap combines minMatchMap and topologyPairsMaps to represent
+// (1) how existing pods match incoming pod on its spread constraints
+// (2) minimum match number of each hard spread constraint
+type topologyPairsPodSpreadMap struct {
```

Comment:
At this moment, I'd prefer to keep it as a standalone data structure because:

1. From the name, we can easily know it's related and only related with `EvenPodsSpread` feature.
2. Keep it standalone so as to make the code more testable.

Reply:
SG